samples/android/video-recorder/src/org/opencv/samples/recorder/RecorderActivity.java

```diff
@@ -0,0 +1,329 @@
+        }
+    }
+
+    public void setErrorStatus() {
+        status = STATUS_ERROR;
+        textView.setText("Status: Error");
+    }
+
+    public boolean startPreview() {
+        mOpenCvCameraView.enableView();
+        mOpenCvCameraView.setVisibility(View.VISIBLE);
+        return true;
+    }
+
+    public boolean startRecording() {
+        Log.i(TAG,"Starting recording");
+
+        File file = new File(videoFilename);
+        file.delete();
+
+        videoWriter = new VideoWriter();
+        if (!useMJPG) {
+            videoWriter.open(videoFilename, Videoio.CAP_ANDROID, VideoWriter.fourcc('H', '2', '6', '4'), FPS, new Size(width, height));
+            if (!videoWriter.isOpened()) {
+                Log.i(TAG,"Can't record H264. Switching to MJPG");
+                useMJPG = true;
+                videoFilename = getFilesDir() + "/" + FILENAME_AVI;
+            }
+        }
+
+        if (useMJPG) {
+            videoWriter.open(videoFilename, VideoWriter.fourcc('M', 'J', 'P', 'G'), FPS, new Size(width, height));
+        }
+
+        Log.d(TAG, "Size: " + String.valueOf(width) + "x" + String.valueOf(height));
+        Log.d(TAG, "File: " + videoFilename);
+
+        if (videoWriter.isOpened()) {
+            Toast.makeText(this, "Record started to file " + videoFilename, Toast.LENGTH_LONG).show();
+            return true;
+        } else {
+            Toast.makeText(this, "Failed to start a record", Toast.LENGTH_LONG).show();
+            return false;
+        }
+    }
+
+    public boolean stopRecording() {
+        Log.i(TAG, "Finishing recording");
+        mOpenCvCameraView.disableView();
+        mOpenCvCameraView.setVisibility(SurfaceView.GONE);
+        videoWriter.release();
+        videoWriter = null;
+        return true;
+    }
+
+    public boolean startPlayback() {
+        imageView.setVisibility(SurfaceView.VISIBLE);
+
+        if (!useMJPG){
+            videoCapture = new VideoCapture(videoFilename, Videoio.CAP_ANDROID);
+        } else {
+            videoCapture = new VideoCapture(videoFilename);
+        }
+        if (!videoCapture.isOpened()) {
+            Log.e(TAG, "Can't open video");
+            Toast.makeText(this, "Can't open file " + videoFilename, Toast.LENGTH_SHORT).show();
+            return false;
+        }
+        Toast.makeText(this, "Starting playback from file " + videoFilename, Toast.LENGTH_SHORT).show();
+        runnable = new Runnable() {
+            @Override
+            public void run() {
+                if (videoCapture == null || !videoCapture.isOpened()) {
+                    return;
+                }
+                Mat frame = new Mat();
+                videoCapture.read(frame);
+                if (frame.empty()) {
+                    if (status == STATUS_PLAYING) {
+                        changeStatus();
+                    }
+                    return;
+                }
+                Imgproc.cvtColor(frame, frame, Imgproc.COLOR_BGR2RGB);
```

Comment:
It should be Imgproc.`COLOR_BGR2RGBA`, because bitmap is created as `Bitmap.Config.ARGB_8888`